src/native/eventpipe/ep.h

```diff
@@ -320,6 +320,28 @@ EP_WRITE_BUFFER_INT (64, u)
 
 #undef EP_WRITE_BUFFER_INT
 
+#if SIZEOF_VOID_P == 4
+# define ep_write_buffer_uintptr_t(buffer, value) ep_write_buffer_uint32_t ((buffer), (uint32_t)(value))
+#else
+# define ep_write_buffer_uintptr_t(buffer, value) ep_write_buffer_uint64_t ((buffer), (uint64_t)(value))
+#endif
+
+static
+inline
+size_t
+ep_write_buffer_string_utf8_to_utf16_t (uint8_t **buf, const ep_char8_t *str, size_t len)
+{
+	ep_char16_t *str_utf16 = ep_rt_utf8_to_utf16le_string (str, len);
```

Comment:
nit: Optimize this for str == NULL or len == 0? It is the current use case for bulk event so it could be called rather frequently called that way.